add zero-shot pipeline

magesh-technovator opened the pull request:
Hi @bhavsarpratik ,
I have added zero-shot classification using the hugging face pipeline.
I don't have much experience with GitHub workflow, just copied the classification.yml and replaced the name and path "classification" -> "zero_shot_classification". Hope this works.

magesh-technovator commented:
Test cases were written based on predictions from "torch-1.8.1", due to version mismatch, there were some round-down errors.
Updated the test case based on "torch-1.8.0".

bhavsarpratik commented:
Hey @magesh-technovator, thanks for such a quick PR.
The run is failing because it is not able to access my AWS creds in your PR. I will look into this.

bhavsarpratik commented on zero_shot_classification/README.MD:
````diff
@@ -0,0 +1,24 @@
+```
+{
+        "texts": ["food was great", "food was bad", "i am going out for food"],
+        "labels": ["negative", "postive", "neutral"],
+        "hypothesis": "The sentiment of the review is {}.",
+        "model_name": "typeform/mobilebert-uncased-mnli",
+        "tokenizer_name": "typeform/mobilebert-uncased-mnli"
````
If these 4 are not passed, it should use the default values.
Also add `multi_label` in config and set it to False. Request can also have non-mandatory `multi_label`

bhavsarpratik commented on .github/workflows/zero_shot_classification.yml:
```diff
@@ -0,0 +1,39 @@
+    - name: Build container
+      run: |
+        docker build --tag zero_shot_classification:latest .
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: us-east-1
+    - name: Push2ECR
+      id: ecr
+      uses: jwalton/gh-ecr-push@v1
+      with:
+        access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        region: us-east-1
+        image: zero_shot_classification:latest
+    - name: Update lambda with image
+      run: aws lambda update-function-code --function-name  zero_shot_classification --image-uri 968911158010.dkr.ecr.us-east-1.amazonaws.com/zero_shot_classification:latest
```
Comment this for now as action is not able to access creds in the PR

bhavsarpratik replied:
Rename to `zero-shot-classification`

bhavsarpratik commented:
I have sent you collaboration invitation. I think the workflow should work if you accept it.

bhavsarpratik commented on zero_shot_classification/src/classifier.py:
```diff
@@ -51,22 +51,32 @@ def __call__(self, request: dict)-> dict:
 
-        model_name = request["model_name"]
-        tokenizer_name = request["tokenizer_name"]
+        labels = request["labels"] if "labels" in request.keys() else config.DEFAULT_CANDIDATE_LABELS
```
This works but a cleaner way is
request.get("labels", config.DEFAULT_CANDIDATE_LABELS)

Please change for all

bhavsarpratik commented on .github/workflows/zero_shot_classification.yml:
```diff
@@ -21,19 +21,19 @@ jobs:
-        image: zero_shot_classification:latest
-    - name: Update lambda with image
-      run: aws lambda update-function-code --function-name  zero_shot_classification --image-uri 968911158010.dkr.ecr.us-east-1.amazonaws.com/zero_shot_classification:latest
+    # - name: Configure AWS Credentials
+    #   uses: aws-actions/configure-aws-credentials@v1
+    #   with:
+    #     aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+    #     aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+    #     aws-region: us-east-1
+    # - name: Push2ECR
+    #   id: ecr
+    #   uses: jwalton/gh-ecr-push@v1
+    #   with:
+    #     access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+    #     secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+    #     region: us-east-1
+    #     image: zero_shot_classification:latest
```
Lets uncomment out from line 24-37 and see if it runs

magesh-technovator commented:
Hi @bhavsarpratik,
I have added the multi_label option (Output format is changed also to support multi-label) and modified the code to work without non-mandatory parameters.

Also, modified test cases for 3 scenarios: complete request, request without default parameters, multi_label.

bhavsarpratik commented on .github/workflows/zero_shot_classification.yml:
```diff
@@ -0,0 +1,39 @@
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./zero_shot_classification
```
Rename dir to `zero-shot-classification` and change it everywhere. All pipeline names will have this format
